### DISCUSSION TITLE
Allow init and set accessor in property

### DISCUSSION THREAD

**Discussion by FaustVX** [dotnet/csharplang#4164] [Language Ideas] [answered]
Hi,

I've came across a situation where I wanted a property to have an init accessor and a private set accessor, to init it (without a constructor) and also allow to modify it from this class.

```csharp
private sealed class WaitAction : Action
{
    public int Ticks { get; init; private set; }

    public override bool MoveNext(Entity entity)
        => --Ticks > 0;
}
```

Then when I wanted to initialize it : `var wait = new WaitAction() { Ticks = 10 };`

Of course, I could write a constructor and make the prop { get; private set; }, or remove the init and make the prop { get; set; }

**Comment by HaloFour** [comment, marked as answer]
I don't think that this would be possible as the `init` accessor already occupies the slot in the runtime for a `set` accessor.  What you could do instead is to explicitly define your backing field and update it directly:

```cs
private sealed class WaitAction : Action
{
    private int ticks;
    public int Ticks {
        get => ticks;
        init => ticks = value;
    }

    public override bool MoveNext(Entity entity)
        => --ticks > 0;
}
```

**Reply by FaustVX** [reply]
> I don't think that this would be possible as the init accessor already occupies the slot in the runtime for a set accessor

Yes, that's what I thought.

So, what I've done is make the set public. It's for a personal project, so there are no real consequences.

Maybe, if this case was found earlier, the dev team could have anticipated this.

**Reply by 333fred** [reply]
> Maybe, if this case was found earlier, the dev team could have anticipated this.

We did consider this use case, and decided that we didn't want to support the scenario.
